src/app/server/DefaultTermsAndConditionsProvider.cpp

```diff
@@ -0,0 +1,306 @@
+        outState = TermsAndConditionsState::REQUIRED_TC_NOT_ACCEPTED;
+        return CHIP_NO_ERROR;
+    }
+
+    // All validation check succeeded...
+    ChipLogProgress(AppServer, "Required terms and conditions, 0x%04x, have been accepted", requiredTermsAndConditions.value);
+    outState = TermsAndConditionsState::OK;
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::CommitAcceptance()
+{
+    VerifyNotNullOrReturnUninitialized(mTermsAndConditionsStorageDelegate);
+
+    CHIP_ERROR err;
+
+    if (!mTemporalAcceptance.HasValue())
+    {
+        ChipLogError(AppServer, "No terms and conditions to commit");
+        return CHIP_NO_ERROR;
+    }
+
+    err = mTermsAndConditionsStorageDelegate->Set(mTemporalAcceptance.Value());
+    if (CHIP_NO_ERROR != err)
+    {
+        ChipLogError(AppServer, "Failed storage delegate Set(): %" CHIP_ERROR_FORMAT, err.Format());
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    ChipLogProgress(AppServer, "Terms and conditions have been committed");
+    mTemporalAcceptance.ClearValue();
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::GetAcceptance(Optional<TermsAndConditions> & outTermsAndConditions) const
+{
+    VerifyNotNullOrReturnUninitialized(mTermsAndConditionsStorageDelegate);
+
+    CHIP_ERROR err;
+
+    // Return the in-memory acceptance state
+    if (mTemporalAcceptance.HasValue())
+    {
+        outTermsAndConditions = mTemporalAcceptance;
+        return CHIP_NO_ERROR;
+    }
+
+    // Otherwise, try to get the persisted acceptance state
+    err = mTermsAndConditionsStorageDelegate->Get(outTermsAndConditions);
+    if (CHIP_ERROR_PERSISTED_STORAGE_VALUE_NOT_FOUND == err)
+    {
+        ChipLogError(AppServer, "No terms and conditions have been accepted");
+        return CHIP_NO_ERROR;
+    }
+
+    // If the storage delegate returns an error, other than "value not found" then there was an unexpected datastore failure
+    if (CHIP_NO_ERROR != err)
+    {
+        ChipLogError(AppServer, "Failed storage delegate Get(): %" CHIP_ERROR_FORMAT, err.Format());
+        return CHIP_ERROR_INTERNAL;
+    }
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::GetRequirements(Optional<TermsAndConditions> & outTermsAndConditions) const
+{
+    VerifyNotNullOrReturnUninitialized(mTermsAndConditionsStorageDelegate);
+
+    outTermsAndConditions = mRequiredAcknowledgements;
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR chip::app::DefaultTermsAndConditionsProvider::ResetAcceptance()
+{
+    VerifyNotNullOrReturnUninitialized(mTermsAndConditionsStorageDelegate);
+
+    CHIP_ERROR err;
+
+    err = mTermsAndConditionsStorageDelegate->Delete();
+    if (CHIP_NO_ERROR != err)
+    {
+        ChipLogError(AppServer, "Failed storage delegate Delete(): %" CHIP_ERROR_FORMAT, err.Format());
+        return CHIP_ERROR_INTERNAL;
```

Comment:
Suggest nor returning an error here, as the client can't do anything with it.

Reply:
OK, updated.